pkg/contexts/ocm/transfer/transfer.go

```diff
@@ -124,65 +139,93 @@ func CopyVersion(printer common.Printer, log logging.Logger, hist common.History
 	*t.GetDescriptor() = *src.GetDescriptor().Copy()
 	log.Info("  transferring resources")
 	for i, r := range src.GetResources() {
-		var m ocm.AccessMethod
-		a, err := r.Access()
-		if err == nil {
-			m, err = r.AccessMethod()
-			if err == nil {
-				defer m.Close()
-				ok := a.IsLocal(src.GetContext())
-				if !ok {
-					if !none.IsNone(a.GetKind()) {
-						ok, err = handler.TransferResource(src, a, r)
-						if !ok {
-							log.Info("transport omitted", "resource", r.Meta().Name, "index", i, "access", a.GetType())
-						}
+		c := &runContext{}
+
+		// gather the access and the access method
+		a := run(c, r.Access)
```

Comment:
it is safe to use the same `runContext` and overwrite the error? maybe join them instead?

Reply:
The purpose is to not do anything if there was an error previously.